src/ORM/Association/DependentDeleteHelper.php

```diff
@@ -45,7 +45,12 @@ public function cascadeDelete(Association $association, EntityInterface $entity,
         /** @psalm-suppress InvalidArgument */
         $foreignKey = array_map([$association, 'aliasField'], (array)$association->getForeignKey());
         $bindingKey = (array)$association->getBindingKey();
-        $conditions = array_combine($foreignKey, $entity->extract($bindingKey));
+        $conditions = array_combine(
+            array_map(function ($field) {
+                return $field . ' IS';
+            }, $foreignKey),
+            $entity->extract($bindingKey)
+        );
```

Comment:
Should this bail out instead of running the query? This would delete all target table rows with foreign key null which might not be desired either.